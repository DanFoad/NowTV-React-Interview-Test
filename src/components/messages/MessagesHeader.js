import React from 'react'
import PropTypes from 'prop-types'

import './Messages.css'

class MessagesHeader extends React.Component {
    getUserAvatars() {
        let avatars = []
        let users = [ ...this.props.users ]
        let length = users.length
        let additional = <span />

        if (users.length > 5) {
            length = 5;
            additional = <span className="messages-header__additional">+{ users.length - 5 }...</span>
        }

        for (var i = 0; i < length; i++) {
            if (users[i].avatar == null) users[i].avatar = "/unknownuser.png"
            avatars.push( <img key={ i } className="messages-header__avatar" src={ users[i].avatar } alt="" /> )
        }

        return (
            <div className="messages-header__avatars">
                { avatars }
                { additional }
            </div>
        )
    }

    getChatTitle() {
        return <h2>{ this.props.chatTitle }</h2>
    }

    render() {
        return (
            <div className="messages-header">
                { this.getUserAvatars() }
                { this.getChatTitle() }
                <div className="messages-header__settings">
                    <img src="/settings.png" alt="" />
                </div>
            </div>
        )
    }
}

MessagesHeader.propTypes = {
    users: PropTypes.array,
}

export default MessagesHeader