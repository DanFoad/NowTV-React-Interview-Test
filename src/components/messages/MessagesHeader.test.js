import React from "react";
import renderer from "react-test-renderer/";

import MessagesHeader from "./MessagesHeader";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MessagesHeader
        users={[
          {
            id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
            firstName: "Albert",
            lastName: "Rose",
            email: "arosec@bbb.org",
            avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
            ip: "20.79.231.223"
          }
        ]}
        chatTitle="Test"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders multiple avatars", () => {
  const tree = renderer
    .create(
      <MessagesHeader
        users={[
          {
            id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
            firstName: "Albert",
            lastName: "Rose",
            email: "arosec@bbb.org",
            avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
            ip: "20.79.231.223"
          },
          {
            id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
            firstName: "Albert",
            lastName: "Rose",
            email: "arosec@bbb.org",
            avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
            ip: "20.79.231.223"
          },
          {
            id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
            firstName: "Albert",
            lastName: "Rose",
            email: "arosec@bbb.org",
            avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
            ip: "20.79.231.223"
          },
          {
            id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
            firstName: "Albert",
            lastName: "Rose",
            email: "arosec@bbb.org",
            avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
            ip: "20.79.231.223"
          }
        ]}
        chatTitle="Test"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("hides further avatars if > 5", () => {
    const tree = renderer
      .create(
        <MessagesHeader
          users={[
            {
              id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
              firstName: "Albert",
              lastName: "Rose",
              email: "arosec@bbb.org",
              avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
              ip: "20.79.231.223"
            },
            {
              id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
              firstName: "Albert",
              lastName: "Rose",
              email: "arosec@bbb.org",
              avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
              ip: "20.79.231.223"
            },
            {
              id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
              firstName: "Albert",
              lastName: "Rose",
              email: "arosec@bbb.org",
              avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
              ip: "20.79.231.223"
            },
            {
              id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
              firstName: "Albert",
              lastName: "Rose",
              email: "arosec@bbb.org",
              avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
              ip: "20.79.231.223"
            },
            {
                id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
                firstName: "Albert",
                lastName: "Rose",
                email: "arosec@bbb.org",
                avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
                ip: "20.79.231.223"
              },
              {
                id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
                firstName: "Albert",
                lastName: "Rose",
                email: "arosec@bbb.org",
                avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
                ip: "20.79.231.223"
              },
              {
                id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
                firstName: "Albert",
                lastName: "Rose",
                email: "arosec@bbb.org",
                avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
                ip: "20.79.231.223"
              },
              {
                id: "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
                firstName: "Albert",
                lastName: "Rose",
                email: "arosec@bbb.org",
                avatar: "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
                ip: "20.79.231.223"
              },
          ]}
          chatTitle="Test"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  