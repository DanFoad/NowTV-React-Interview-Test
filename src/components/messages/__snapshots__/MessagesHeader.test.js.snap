exports[`test hides further avatars if > 5 1`] = `
<div
  className="messages-header">
  <div
    className="messages-header__avatars">
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <span
      className="messages-header__additional">
      +
      3
      ...
    </span>
  </div>
  <h2>
    Test
  </h2>
  <div
    className="messages-header__settings">
    <img
      alt=""
      src="/settings.png" />
  </div>
</div>
`;

exports[`test renders correctly 1`] = `
<div
  className="messages-header">
  <div
    className="messages-header__avatars">
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <span />
  </div>
  <h2>
    Test
  </h2>
  <div
    className="messages-header__settings">
    <img
      alt=""
      src="/settings.png" />
  </div>
</div>
`;

exports[`test renders multiple avatars 1`] = `
<div
  className="messages-header">
  <div
    className="messages-header__avatars">
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <img
      alt=""
      className="messages-header__avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff" />
    <span />
  </div>
  <h2>
    Test
  </h2>
  <div
    className="messages-header__settings">
    <img
      alt=""
      src="/settings.png" />
  </div>
</div>
`;
